common/djangoapps/entitlements/models.py

```diff
@@ -48,7 +50,8 @@ class CourseEntitlementPolicy(models.Model):
                    "it is no longer able to be regained by a user."),
         null=False
     )
-    site = models.ForeignKey(Site)
+    site = models.ForeignKey(Site, null=True)
+    mode = models.CharField(max_length=32, choices=MODES, default=DEFAULT_ENROLLMENT_MODE, null=True)
```

Comment:
See comment above about defaulting the Mode.

Reply:
Also see [previous response](https://github.com/edx/edx-platform/pull/17758#discussion_r177746312)